src/components/ListItem.jsx

```diff
@@ -1,5 +1,105 @@
 import './ListItem.css';
+import { updateItem } from '../api/firebase';
+import { useState, useEffect } from 'react';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ name, itemId, dateLastPurchased }) {
+	const [check, setCheck] = useState(false);
+
+	useEffect(() => {
+		const currentDate = new Date();
+		const purchasedDate = dateLastPurchased
+			? dateLastPurchased.toDate()
+			: dateLastPurchased;
+		const timeElapsed = Math.abs(currentDate - purchasedDate);
+		const hoursElapsed = timeElapsed / (1000 * 60 * 60);
```

Comment:
Try logging out the results you get here. JavaScript and large integers don't play particularly well together, so there may be some unexpected results.